librz/search/string_search.c

```diff
@@ -37,23 +36,24 @@ static bool string_find(RZ_NULLABLE RzSearchFindOpt *fopt, void *user, ut64 offs
 		void **it_m = NULL;
 		rz_pvector_foreach (ss->strings, it_m) {
 			RzDetectedString *find = *it_m;
-			if (detected->length < find->length) {
-				// ignore strings that are smaller than the one we are looking for.
-				continue;
-			}
-			size_t len = RZ_MIN(detected->length, find->length);
-			if ((ss->caseless && rz_str_ncasecmp(detected->string, find->string, len)) ||
-				(!ss->caseless && strncmp(detected->string, find->string, len))) {
-				// ignore strings that are not matching till len.
-				continue;
-			}
-
-			RzSearchHit *hit = rz_search_hit_new("string", detected->addr, detected->size);
-			if (!hit || !rz_th_queue_push(hits, hit, true)) {
-				rz_search_hit_free(hit);
-				rz_list_free(found);
-				return false;
+			RzPVector *matches = rz_regex_match_all(find->regex, detected->string, detected->size, 0, RZ_REGEX_DEFAULT);
```

Comment:
you forgot to pass the flags.

Reply:
Those are only flags used during compilation. Added docs for this.